### ISSUE TITLE
* is deleted in right outer join request

### ISSUE THREAD

**Issue by sgaunet** [sqlc-dev/sqlc#1995]
### Version

Other

### What happened?

Hello,

I'm using version 1.16.0

### Relevant log output

```shell
I've got an error when executing the request: pq: column "ts" does not exist
```

### Database schema

```sql
Not relevant, see SQL queries
```

### SQL queries

```sql
SQL request:

-- name: CountAlertReportBy :many
select DATE_TRUNC($1,ts)::text as datetime,coalesce(count,0) as count from 
(
    SELECT  DATE_TRUNC($1,eventdate) as hr ,count(*)
    FROM    alertreport
    where eventdate between $2 and $3
    GROUP BY   1
) AS cnt 
right outer join ( SELECT * FROM generate_series ( $2, $3, CONCAT('1 ',$1)::interval) AS ts ) as dte
on DATE_TRUNC($1, ts ) = cnt.hr 
order by 1 asc


The generated version: 

const countAlertReportBy = `-- name: CountAlertReportBy :many
select DATE_TRUNC($1,ts)::text as datetime,coalesce(count,0) as count from 
(
    SELECT  DATE_TRUNC($1,eventdate) as hr ,count(*)
    FROM    alertreport
    where eventdate between $2 and $3
    GROUP BY   1
) AS cnt 
right outer join ( SELECT  FROM generate_series ( $2, $3, CONCAT('1 ',$1)::interval) AS ts ) as dte
on DATE_TRUNC($1, ts ) = cnt.hr 
order by 1 asc
`

The asterisk is missing here => right outer join ( SELECT  FROM generate_series ( $2, $3, CONCAT('1 ',$1)::interval) AS ts ) as dte

It causes the error
```

### What operating system are you using?

Linux

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by sgaunet** [comment]
Hi,

I can try to fix this error, at least I would like to try. I see the endtoend tests and will add my case. But first of all, it should pass without any modifications and I have this error when running endtoend tests:

```
--- FAIL: TestReplay (0.06s)
    --- FAIL: TestReplay/testdata/process_plugin_sqlc_gen_json (0.00s)
        /home/sylvain/OSS/sqlc/internal/endtoend/endtoend_test.go:118: sqlc generate failed: # package jsonb
            error generating code: process: sqlc-gen-json not found

```
Could you tell me what's wrong with my env ? (I guess)

Thanks

**Comment by sgaunet** [comment]
I've found a workaround:

```
-- name: CountAlertReportBy :many
select DATE_TRUNC(sqlc.arg(period),ts)::text as datetime,coalesce(count,0) as count from 
(
    SELECT  DATE_TRUNC(sqlc.arg(period),eventdate) as hr ,count(*)
    FROM    alertreport
    where eventdate between sqlc.arg(dateBegin) and sqlc.arg(dateEnd)
    GROUP BY   1
) AS cnt 
right outer join ( SELECT generate_series ( sqlc.arg(dateBegin), sqlc.arg(dateEnd), CONCAT('1 ',sqlc.arg(period))::interval) AS ts ) as dte
on DATE_TRUNC(sqlc.arg(period), ts ) = cnt.hr 
order by 1 asc;
```

It works as expected if we remove '* FROM'.

**Comment by andrewmbenton** [comment]
Sorry for the delay here. I'm presuming you've moved on but just in case this is still relevant, the test failure you saw is due to an assumption in an older version of the test suite regarding the presence of a binary named `sqlc-gen-json` on your `$PATH`. That assumption is now removed, and all tests should pass without modification.

Regarding the underlying issue, it likely remains, so I'll leave this open even though you found a workaround for your specific case.

**Comment by mafredri** [comment]
I believe this issue could be distilled into: _sqlc doesn't handle `FROM generate_series`, only `SELECT generate_series`_.

I have the following query:

```sql
SELECT d FROM generate_series('2023-07-18T00:00:00Z'::timestamptz, '2023-07-20T00:00:00Z'::timestamptz, '5 minute'::interval) d
````

And we get:

```
query.sql:2:3: column "d" does not exist
```

It should work, but `sqlc` doesn't seem to understand that `FROM generate_series` produces rows with one column, and that we can alias it.

This can be worked around via `SELECT`, e.g. as follows:

```sql
WITH d AS (
	SELECT generate_series('2023-07-18T00:00:00Z'::timestamptz, '2023-07-20T00:00:00Z'::timestamptz, '5 minute'::interval) AS d
)
SELECT d FROM d
```

Apologies if you feel this is not related enough to be in the same issue, I can report it in a separate one if that's preferable.

**Comment by vegather** [comment]
This can also be worked around by doing something with the resulting column, for example adding something to it.

This query works:
```sql
SELECT i + 0 FROM generate_series(0, 10) AS i;
```

while this results in a `column "i" does not exist` error:
```sql
SELECT i FROM generate_series(0, 10) AS i;
```

**Comment by mafredri** [comment]
Oh, that's an interesting workaround @vegather, thanks for sharing it. Seems it's enough to add a type cast (`name::type`) to avoid the error! This will work great for my use-case.

I think it'd be great if the error message could at least be a bit more helpful, or perhaps the "computation" requirement can be removed all-together?

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/star_expansion_series/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/4b610763d6d66a4b50c63b5ea1fa22567899c2c1de1278b5b24da627fab56f94)

**Comment by sgaunet** [comment]
Thank you :)
